src/nupic/math/ArrayAlgo.hpp

```diff
@@ -5241,7 +5142,7 @@ namespace nupic {
                      : 
                      );
 
-  #elif defined(NTA_ARCH_64)
+  #else //64bit
```

Comment:
Right, it's a matter of taste. I read somewhere (and prefer) to set the least limiting guards (most specific). What you say about the new arch is true, on the other hand, eg (I think @oxtopus ) made port to Solaris which defines a specific platform (NTA_OS_Solaris64) ..and it will take advantage from this too (as Solaris is UNIX).